server/routes/charts.js

```diff
@@ -189,4 +189,28 @@ router
       return res.status(500).json({ message: error.message })
     }
   })
+router
+  .route('/api/v1/charts/duplicate')
+  .post(ensureAuthenticated, async (req, res) => {
+    try {
+      const { dataDb } = req.app.locals
+      const { chart_id } = req.body
+      const sourceChart = await dataDb.collection(collections.charts).findOne(
+        {
+          _id: new ObjectID(chart_id),
+        },
+        { projection: { _id: 0 } }
+      )
+      const { insertedId } = await dataDb
+        .collection(collections.charts)
+        .insertOne({
+          ...sourceChart,
+          title: sourceChart.title + ' Duplicate',
+        })
+
+      return res.status(200).json({ data: insertedId })
+    } catch (error) {
+      return res.status(404).json({ message: 'Chart was not duplicated' })
+    }
+  })
```

Comment:
Looks good!
This might be a concern for another day -- Do we need any kind of control around which charts a user can access? Like is there anything to stop a user from submitting some random chart_id that they don't have access too ( if that's a thing ) and creating a duplicate that they then do have access to?

Reply:
haha that's something I was thinking about for this new feature. It's something that will be coming soon.